File: test-matmul.py
import numpy as np

if __name__ == "__main__":
  A = np.matrix([
    [1,2,3],
    [4,5,6],
    [7,8,9]
  ])
  print(f"input A:\n{A}\n")
  B = np.matrix([
    [1],
    [2],
    [3]
  ])
  print(f"input B:\n{B}\n")
  C = A @ B
  print(f"output:\n{C}\n")

File: test-matmul.cpp
#include "./lib/matrix.hpp"
#include <iostream>

int main() {
	using namespace std;
	using namespace feng;

	matrix<float> m { 3, 3 };
	int val = 1;
	for (int i = 0; i < 3; i++) {
		for (int j = 0; j < 3; j++) {
			m[i][j] = val;
			val++;
		}
	}

	cout << "input A:" << endl;
	cout << m << endl;

	matrix<float> n { 3, 1 };
	n[0][0] = 1;
	n[1][0] = 2;
	n[2][0] = 3;

	cout << "input B:" << endl;
	cout << n << endl;

	matrix<float> o = m * n;

	cout << "output:" << endl;
	cout << o << endl;
}
